### ISSUE TITLE
[wish] use local `README.md`when off-line

### ISSUE THREAD

**Issue by retokromer** [amiaopensource/homebrew-amiaos#225]
I suggest to modify the call of the [documentation](https://github.com/amiaopensource/vrecord/blob/ebc8b142fa30f7ddb0624326a0625b475e72ea80/vrecord#L194) like we did e.g. for [ffmprovisr](https://github.com/amiaopensource/ffmprovisr/blob/58aa0549ff607f3cfbcb0d92404b5959598e9c5c/scripts/ffmprovisr#L11):
- if on-line, then show the version on GitHub
- if off-line, then use the version installed on the computer

**Comment by dericed** [comment]
Actually, there's an option to transfer an issue in github, so I moved it.

**Comment by retokromer** [comment]
If I don’t miss something, we need to adapt also the the call of the documentation in the `vrecord` code itself. I have proposed one possible way at the beginning of this issue. (FYI: Quite often I do use the off-line versions of `ffmprovisr` and `cable-bible` for training purposes and they work well.)

**Comment by privatezero** [comment]
The issue `ffmprovisr` and `cable-bible` are both written in HTML, and the vrecord documentation is written in a distributed series of markdown files. This renders nicely online, but not so much locally. So yes, the vrecord documentation call would need to be adapted, but that is the relatively easy part of the process, and one that is dependent on a satisfactory local render of the docs being made via the install formula. (Can't point the vrecord document call at something if there aren't local docs to point at, and the series of markdown files is pretty useless in that context).

If my current attempt within this repo of trying to use `cat` and `sed` piped into Pandoc to create a usable local HTML file is satisfactory (at least for now), then I could update and merge it and we could make the tweaks within the vrecord code. This would, however, slightly expand vrecord's dependencies.

**Comment by retokromer** [comment]
I was thinking of improving the [documentation call](https://github.com/amiaopensource/vrecord/blob/373425e3c17e2e6f1779dd3688df2622470cc029/vrecord#L497-L499) provided by the GUI.

**Comment by privatezero** [comment]
@retokromer are you suggesting to have the documentation call concatenate the markdown on the fly every time the documents are accessed? I would think that having a usable html file created on install by the ruby script, that then could be accessed by vrecord if the online docs were unavailable would be more efficient?

**Comment by retokromer** [comment]
@privatezero I wasn’t suggesting to use markdown, but indeed to create a HTML documentation at the installation, like the prototype you already made with `pandoc`. However, I would suggest to add at the documentation call in `vrecord` a test:
- if on-line, then show the version on GitHub;
- if off-line, then use the local version installed on the computer.

Of course, this increases `vrecord`’s dependencies.

**Comment by privatezero** [comment]
Ah - yeah, that is what I was thinking as well. Once we get a solid doc install we can incorporate it into a test

**Comment by retokromer** [comment]
@privatezero Are you working on https://github.com/amiaopensource/homebrew-amiaos/tree/changes-for-doc-install  (il looks stalled since quite a while, therefore I am asking)?

**Comment by privatezero** [comment]
Hi @retokromer, haven't had time to work on this - not actively working on it right now. I still haven't had any ideas for a more elegant solution for handling the broken up markdown files than the one in that branch, which isn't particularly elegant. Any thoughts?

**Comment by retokromer** [comment]
@privatezero Thank you for the information! Then I will delve into it once the Bash webinars are over.

**Comment by retokromer** [comment]
@privatezero @dericed Is currently the documentation in synch with the program?

**Comment by privatezero** [comment]
Hi @retokromer no, it hasn't kept pace with many new options and menus unfortunately. Had been hoping to put out a call for help at the AMIA hackday for this, but that has recently been postponed by the organizers. I guess it is probably best to update sooner than later and not save that work for a rescheduled hackday.

**Comment by retokromer** [comment]
@privatezero Thank you for confirming my feelings! I will wait, before trying to set up a pandoc mixing.

**Comment by privatezero** [comment]
Hi @retokromer, were you no longer interested in this? I personally haven't found any elegant solutions to getting all of the individual markdown files compiled together and then translated into something more user friendly than raw markdown. Did some experiments with wget as well, but was running into issues around internal links and markdown style sheets

**Comment by retokromer** [comment]
Hello, @privatezero! Personally I am still interested, but I opened this issue almost… three years ago. I also suggested to restructure the package, what would made things simpler, in my opinion. On my end, I have reactivated the boring solution of patching the releases. That way I can work with the current Bash which makes things simpler. I am still using `pandoc`, inspired by your working branch, which gives at least “something”.
